src/relations/postgresql_provider.py

```diff
@@ -57,6 +62,9 @@ def __init__(self, charm: CharmBase, relation_name: str = "database") -> None:
             charm.on[self.relation_name].relation_changed,
             self._on_relation_changed_event,
         )
+        self.framework.observe(
+            charm.on[self.relation_name].relation_created, self.register_address_for_endpoint
+        )
```

Comment:
Probably we also need to run that method in the config changed event to update the IP address when it changes as per https://documentation.ubuntu.com/juju/3.6/reference/hook/#config-changed:

> config-changed
It also runs whenever:
when any IP address of the machine hosting the unit changes.